pkg/gateway/kubernetes.go

```diff
@@ -6,8 +6,11 @@ import (
 	kclient "k8s.io/kubernetes/pkg/client/unversioned"
 	kclientcmd "k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
 	kclientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	"log"
 )
 
+const VanityConfigMapName = "FarvaVanityUrls"
```

Comment:
1. I'd prefer not to leak the "farva" piece out into Kubernetes land if possible. 
2. This should be a general-purpose config so we aren't screwed when we want to add more data to this in the future.

I'd suggest the default naming as `gateway-config`, but make this configurable via flag.